packages/front-end/components/Settings/ClickToCopy.tsx

```diff
@@ -0,0 +1,42 @@
+import clsx from "clsx";
+import { ReactNode, useState } from "react";
+import Tooltip from "../Tooltip/Tooltip";
+import styles from "./ClickToCopy.module.scss";
+
+type Props = {
+  valueToCopy?: string;
+  children?: ReactNode;
+};
+
+export default function ClickToCopy({ valueToCopy, children }: Props) {
+  const [copyText, setCopyText] = useState("Copy");
+  return (
+    <Tooltip
+      className={clsx(!valueToCopy && styles.blurText, styles.tooltip)}
```

Comment:
It doesn't feel like the blur logic should be in the `ClickToCopy` component.  We should only ever render this component when there is text available to copy.

I think it makes better sense to move most of this logic into the `ClickToReveal` component.  If the value isn't set yet, just return the blurred text and button (no need for a tooltip or ClickToCopy component).  Once the value is set, render the ClickToCopy component instead.

If we do this, then we can get rid of all the conditional logic in this component.  `valueToCopy && ...`